tracer/test/Datadog.Trace.Tools.dd_dotnet.ArtifactTests/CreatedumpTests.cs

```diff
@@ -581,17 +583,38 @@ void ValidateStacktrace(JToken callstack)
                         age.Should().Be(unchecked((uint)pdbInfo.Age));
                         hash.Should().Equal(pdbInfo.Guid.ToByteArray());
                     }
+                    else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                    {
+                        // Validate sofile
+                        if (frame["normalized_ip"] == null)
+                        {
+                            // On linux we can face cases where the build_id is not available:
+                            // - specifically on alpine, /lib/ld-musl-XX do not have a build_id.
+                            // - We are looking at a frame for which the library was unloaded /memfd:doublemapper (deleted)
+                            continue;
+                        }
+
+                        var elfNode = frame["normalized_ip"]["meta"]["Elf"];
+                        var buildId = ((JArray)elfNode["build_id"]).Select(g => g.Value<byte>()).ToArray();
+
+                        using var elf = ELFReader.Load(moduleName);
+                        var buildIdNote = elf.GetSection(".note.gnu.build-id") as INoteSection;
+                        buildId.Should().Equal(buildIdNote.Description);
+                    }
                 }
+            }
 
-                validatedModules.Should().NotBeEmpty();
+            validatedModules.Should().NotBeEmpty();
 
 #if NETFRAMEWORK
-                var clrModuleName = "clr.dll";
+            var clrModuleName = "clr.dll";
 #else
-                var clrModuleName = "coreclr.dll";
+            var clrModuleName = RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ? "libcoreclr.so" : "coreclr.dll";
 #endif
 
-                validatedModules.Should().ContainMatch($@"*\{clrModuleName}");
+            if (!Utils.IsAlpine())
```

Comment:
Why !IsAlpine? 🤔

Reply:
I think the unwinding failed on that test and we did not get to frames from libcoreclr and other.
I added this to my note to check why it's failing and if it's already fixed upstream.
(I pushed fixes months ago)